src/mlpack/core/util/io.cpp

```diff
@@ -267,3 +267,52 @@ void IO::ClearSettings()
   GetSingleton().aliases = persistentAliases;
   GetSingleton().functionMap = persistentFunctions;
 }
+
+void IO::CheckInputMatrices()
+{
+  typedef typename std::tuple<data::DatasetInfo, arma::mat> TupleType;
+  std::map<std::string, util::ParamData>::iterator itr;
+
+  for (itr = IO::Parameters().begin(); itr != IO::Parameters().end(); ++itr)
+  {
+    std::string paramName = itr->first;
+    std::string paramType = itr->second.cppType;
+    std::string errMsg = "The input " + paramName + " has NaN values.";
+    if (paramType == "arma::mat")
+    {
+      if (IO::GetParam<arma::Mat<double>>(paramName).has_nan())
```

Comment:
We should probably check for `Inf` too---you could use `.is_finite()` for that.  http://arma.sourceforge.net/docs.html#is_finite :+1:

Reply:
Yes it is a good idea, I have added the same. I decided to `.has_inf()` just for uniformity with `.has_nan()`. I have also added separate tests for the `nan` and `inf` values in `src/mlpack/bindings/python/tests/test_python_binding.py`

Reply:
@NippunSharma maybe now it makes sense to split out the body of each if statement into a templated function?  The bodies of each if statement are basically identical, so it would be great if we could capture that identical functionality in one standalone function.

Reply:
Yes, that will reduce the repetitive code. I have added `CheckInputMatrix()` function that is called inside the `CheckInputMatrices()` function.